src/Analysis/Ast/Impl/Diagnostics/DiagnosticsEntry.cs

```diff
@@ -51,6 +54,20 @@ public sealed class DiagnosticsEntry {
         /// </summary>
         public DiagnosticSource Source { get; }
 
+        public bool ShouldReport(IPythonModule module) {
+            // Only report for user written modules
+            if (module.ModuleType != ModuleType.User) {
+                return false;
+            }
+
+            // If user specifies #noqa, then do not report diagnostic
+            if (module.GetComment(SourceSpan.Start.Line).EqualsIgnoreCase("noqa")) {
```

Comment:
Reminder to myself to add a null check here.

Reply:
Don't have to if you return `string.Empty`.